### ISSUE TITLE
RemoteSigner must not panic

### ISSUE THREAD

**Issue by alpe** [tendermint/tendermint#3093]
**Tendermint version** (use `tendermint version` or `git rev-parse --verify HEAD` if installed from source):
v0.27.3

**ABCI app** (name for built-in, URL for self-written if it's publicly available):
cosmos-sdk

**Environment**:
- Linux

**What happened**:

Due to network issues/ downtime the RemoteSigner had run into timeouts. This resulted in a `CONSENSUS FAILURE!!!`. 
The problem is with with the panics in:
* [`GetAddress()`](https://github.com/tendermint/tendermint/blob/master/privval/remote_signer.go#L39) or 
* [`GetPubKey()`](https://github.com/tendermint/tendermint/blob/master/privval/remote_signer.go#L49) 

**What you expected to happen**:
Errors should be handled instead. Distributed systems can fail.

**Have you tried the latest version**: yes/no

**How to reproduce it** (as minimally and precisely as possible):

**Logs (paste a small part showing an error (< 10 lines) or link a pastebin, gist, etc. containing more of the log file)**:
```
E[5016-01-05|22:21:56.699] CONSENSUS FAILURE!!!                         module=consensus err="remote signer timed out" stack="goroutine 454 [running]:\nruntime/debug.Stack(0xc00340f660, 0xef8fc0, 0xc00570d300)\n\t/usr/local/go/src/runtime/debug/stack.go:24 +0xa7\ngithub.com/cosmos/cosmos-sdk/vendor/github.com/tendermint/tendermint/consensus.(*ConsensusState).receiveRoutine.func2(0xc0009e8700, 0x113a8e0)\n\t/go/src/github.com/cosmos/cosmos-sdk/vendor/github.com/tendermint/tendermint/consensus/state.go:576 +0x57\npanic(0xef8fc0, 0xc00570d300)\n\t/usr/local/go/src/runtime/panic.go:513 +0x1b9\ngithub.com/cosmos/cosmos-sdk/vendor/github.com/tendermint/tendermint/privval.(*RemoteSignerClient).GetAddress(0xc000feae60, 0x0, 
```
**Config (you can paste only the changes you've made)**:

**node command runtime flags**:

**`/dump_consensus_state` output for consensus bugs**

**Anything else we need to know**:

**Comment by Liamsi** [comment]
Probably a duplicate of #2926 and should be resolved with #2948 (merged to develop, will be contained in next release). `GetAddress` was removed and `GetPublicKey` doesn't panic anymore: https://github.com/tendermint/tendermint/blob/616c3a4baeb1f52589f6afe09bc2be256b9626d2/privval/remote_signer.go#L39-L49

**Comment by alpe** [comment]
Yes, it is duplicate. Sorry.
